atom_gems/AtomTutorials/Code/CMakeLists.txt

```diff
@@ -0,0 +1,205 @@
+if(PAL_TRAIT_BUILD_HOST_TOOLS)
+    # The AtomTutorials.Editor.API target can be used by other gems that want to interact with the AtomTutorials.Editor module
+    ly_add_target(
+        NAME AtomTutorials.Editor.API INTERFACE
+        NAMESPACE Gem
+        FILES_CMAKE
+            atomtutorials_editor_api_files.cmake
+            ${pal_dir}/atomtutorials_editor_api_files.cmake
+        INCLUDE_DIRECTORIES
+            INTERFACE
+                Include
+        BUILD_DEPENDENCIES
+            INTERFACE
+                AZ::AzToolsFramework
+    )
+
+    # The AtomTutorials.Editor.Private.Object target is an internal target
+    # which is only to be used by this gems CMakeLists.txt and any subdirectories
+    # Other gems should not use this target
+    ly_add_target(
+        NAME AtomTutorials.Editor.Private.Object STATIC
+        NAMESPACE Gem
+        AUTOMOC
+        AUTORCC
+        FILES_CMAKE
+            atomtutorials_editor_private_files.cmake
+        TARGET_PROPERTIES
+            O3DE_PRIVATE_TARGET TRUE
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                Include
+                Source
+        BUILD_DEPENDENCIES
+            PUBLIC
+                AZ::AzToolsFramework
+                $<TARGET_OBJECTS:Gem::AtomTutorials.Private.Object>
+    )
+
+    ly_add_target(
+        NAME AtomTutorials.Editor GEM_MODULE
+        NAMESPACE Gem
+        AUTOMOC
+        FILES_CMAKE
+            atomtutorials_editor_shared_files.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                Source
+            PUBLIC
+                Include
+        BUILD_DEPENDENCIES
+            PUBLIC
+                Gem::AtomTutorials.Editor.API
+            PRIVATE
+                Gem::AtomTutorials.Editor.Private.Object
+    )
+
+    # By default, we will specify that the above target AtomTutorials would be used by
+    # Tool and Builder type targets when this gem is enabled.  If you don't want it
+    # active in Tools or Builders by default, delete one of both of the following lines:
+    ly_create_alias(NAME AtomTutorials.Tools    NAMESPACE Gem TARGETS Gem::AtomTutorials.Editor)
+    ly_create_alias(NAME AtomTutorials.Builders NAMESPACE Gem TARGETS Gem::AtomTutorials.Editor)
+
+    # For the Tools and Builders variants of AtomTutorials Gem, an alias to the AtomTutorials.Editor API target will be made
+    ly_create_alias(NAME AtomTutorials.Tools.API NAMESPACE Gem TARGETS Gem::AtomTutorials.Editor.API)
+    ly_create_alias(NAME AtomTutorials.Builders.API NAMESPACE Gem TARGETS Gem::AtomTutorials.Editor.API)
+
+endif()
+
+################################################################################
+# Tests
+################################################################################
+# See if globally, tests are supported
+if(PAL_TRAIT_BUILD_TESTS_SUPPORTED)
+    # We globally support tests, see if we support tests on this platform for AtomTutorials.Tests
+    if(PAL_TRAIT_ATOMTUTORIALS_TEST_SUPPORTED)
```

Comment:
We should be able to just remove the ATOMTUTORIALS specific traits and rely on the more general PAL traits.